src/pages/index.tsx

```diff
@@ -1,19 +1,21 @@
+import { useState } from "react";
+import { useRouter } from "next/router.js";
 import { Stack, Typography } from "@mui/material";
 import { signIn } from "next-auth/react";
-import { useRouter } from "next/router.js";
-
-import { text } from "~/assets/text/index.js";
-import { TopRanked } from "~/components/index.js";
-import { api } from "~/utils/api.js";
-import { pages } from "~/router.js";
-import { EMPTY_RES } from "~/constants/index.js";
+import { Navbar, TopRanked } from "~/components/index.js";
 import { PlayButton } from "~/components/play-button/index.js";
 import { PixelButton } from "~/components/pixel-button/index.js";
 import { BotBusterLogoAnimation } from "~/components/bot-buster-logo/index.js";
+import { LandingPageAnimation } from "~/components/landing-page-animation/index.js";
+import { api } from "~/utils/api.js";
+import { pages } from "~/router.js";
+import { EMPTY_RES } from "~/constants/index.js";
+import { text } from "~/assets/text/index.js";
 import { styles } from "~/styles/pages/homepage.js";
 
 const Homepage = () => {
   const { push } = useRouter();
+  const [menuIsOpen, setMenuIsOpen] = useState(false);
```

Comment:
Why put this logic here and not in the navbar component itself?

Reply:
I built the navbar this way because it was intuitive to me for the parent component to control whether the main-menu dialog would be rendered or not. 
But I suppose we could save on the extra prop and control the open state within the navbar, since I don't think the parent component needs to know about it?

Reply:
Hmm, good point, let's try keep the control states local to the navbar